src/osemgrep/cli_scan/Scan_CLI.ml

```diff
@@ -841,9 +841,18 @@ let cmdline_term ~allow_empty_config : conf Term.t =
     (* ugly: call setup_logging ASAP so the Logs.xxx below are displayed
      * correctly *)
     Logs_.setup_logging ~force_color ~level:common.CLI_common.logging_level ();
-
-    let target_roots = target_roots |> Fpath_.of_strings in
-
+    let git_repo_target =
+      let is_git_repo target =
+        target |> Git_wrapper.remote_repo_name |> Option.is_some
+      in
+      match target_roots with
+      | [ target ] when is_git_repo target -> Some target
+      | target :: _ when is_git_repo target ->
+          Error.abort
+            "Multiple targets are not supported when scanning a remote git \
+             repository"
+      | _ -> None
+    in
```

Comment:
I haven't fully thought this through but it seems to me that a remote repo URL defines a project rather than a target. It should be possible to specify targets within that remote repo. I'm not sure how. Perhaps add an option `--remote REMOTE_URL` and then have all the target paths be interpreted as relative to the root of the remote project (locally some temp folder).

Reply:
I think that makes a lot of sense, --remote + normal targeting seems a lot more sane than having a single URL as a target

Reply:
Yea ended up going with this, it's a lot cleaner and enables this feature for pro things, and for additional targeting